dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTableValue.java

```diff
@@ -152,11 +153,22 @@ public String getDirectoryName() {
     }
 
     private String getParentDir(final String fullName) {
-        final int index = fullName.lastIndexOf('/');
+        String name = fullName;
+        if(name.endsWith("/")) {
+            name = name.substring(0, name.length() - 1);
+        }
+        final int index = name.lastIndexOf('/');
         if (index < 0) {
             return StringConstants.EMPTY_STRING;
         }
-        return fullName.substring(0, index) + '/';
+        final String sub = name.substring(0, index);
+        if (sub.equals(name)) {
+            return "";
```

Comment:
What's the difference between return "" and return StringConstants.EMPTY_STRING?